### PR TITLE
[Backport] 8241040: Support for AVX-512 Ternary Logic Instruction.

### PR DESCRIPTION
**Pull request by JinZhonghui**
Summary: Backport 8241040: A new pass has been added which folds expression tree involving vector boolean logic operations into a MacroLogic node.

Test Plan: ci jtreg

Reviewed-by: JoshuaZhuwj

Issue: https://github.com/alibaba/dragonwell11/issues/367

### CODE REVIEW

**Comment by JinZhonghui** [comment]
@JoshuaZhuwj Please help to review this.

File: src/hotspot/cpu/x86/x86.ad [hunk 1]
```diff
@@ -1513,6 +1513,10 @@ const bool Matcher::match_rule_supported(int opcode) {
         return false; // 128bit vroundpd is not available
       }
       break;
+    case Op_MacroLogicV:
+      if (UseAVX < 3 || !UseVectorMacroLogic) {
+        return false;
+      }
```
Comment (JoshuaZhuwj) [inline src/hotspot/cpu/x86/x86.ad:1519]: add "break;" before next "case"

File: src/hotspot/share/opto/matcher.cpp [hunk 2]
```diff
@@ -2325,7 +2335,7 @@ void Matcher::find_shared( Node *n ) {
       case Op_FmaD:
       case Op_FmaF:
       case Op_FmaVD:
-      case Op_FmaVF: {
+      case Op_FmaVF:{
         // Restructure into a binary tree for Matching.
```
Comment (JoshuaZhuwj) [inline src/hotspot/share/opto/matcher.cpp:2339]: No need to change.

**Review by JoshuaZhuwj — COMMENTED** [review]
See my comments inline.
I think we'd better also introduce the jtreg test sources of original commit.

File: src/hotspot/cpu/x86/x86.ad (hunk 1 above)
Comment (JinZhonghui) [reply src/hotspot/cpu/x86/x86.ad:1519]: Fixed

File: src/hotspot/share/opto/matcher.cpp (hunk 2 above)
Comment (JinZhonghui) [reply src/hotspot/share/opto/matcher.cpp]: Fixed.

**Review by JoshuaZhuwj — APPROVED** [review]
Looks good. Thanks.

**Pull request merged**